### PR TITLE
adding hubspot upload allow type limit

### PR DESCRIPTION
**Pull request by iruzevic**
# Description

### Added
- New global and local settings for limiting HubSpot file upload mime type.

### CODE REVIEW

File: src/Validation/SettingsCaptcha.php [hunk 1]
```diff
@@ -212,7 +212,7 @@ public function getSettingsGlobalData(): array
 						'component' => 'input',
 						'inputId' => $this->getSettingsName(self::SETTINGS_CAPTCHA_SCORE_KEY),
 						'inputFieldLabel' => \__('"Spam unlikely" threshold', 'eightshift-forms'),
-						'inputFieldHelp' => \__('This number determines the level above which a submission is <strong>not</strong> considered spam. <br /> <br /> The value should be between 0.0 and 1.0 (default is 0.5).', 'eightshift-forms'),
+						'inputFieldHelp' => \__('This number determines the level above which a submission is <strong>not</strong> considered spam. In general normal users will always get score of 0.8-0.9. The value should be between 0.1 and 1.0 (default is 0.5).', 'eightshift-forms'),
```
Comment (mbmjertan) [inline src/Validation/SettingsCaptcha.php:215]:
```suggestion
						'inputFieldHelp' => \__('This number determines the level above which a submission is <strong>not</strong> considered spam. In general, normal users will get a score of around 0.8-0.9. The value should be between 0.1 and 1.0 (default is 0.5).', 'eightshift-forms'),
```

**Review by mbmjertan — APPROVED** [review]
Looks good to me!

File: src/Blocks/components/form/assets/form.js [hunk 2]
```diff
@@ -1047,26 +1047,47 @@ export class Form {
...
+		const searchParams = new URLSearchParams(window.location.search);
+		for (const [key, value] of searchParams.entries()) {
+			// Bailout if not allowed or empty
+			if (!allowedTags.includes(key) || value === '') {
+				continue;
+			}
+
+			// Add valid tag.
+			newStorage[key] = value;
+		}
```
Comment (volfkarlo) [inline src/Blocks/components/form/assets/form.js:1082]:
```suggestion
		const newStorage = searchParams.entries()
			.filter(([key, value]) => allowedTags.includes(key) && value !== '')
```

File: src/Blocks/components/form/assets/form.js [hunk 3]
```diff
@@ -1047,26 +1047,47 @@ export class Form {
...
+		// Bailout if nothing is set from allowed tags or everything is empty.
+		if (Object.keys(newStorage).length === 0) {
+			return;
+		}
+
+		// Add current timestamp to new storage.
+		newStorage.timestamp = Date.now();
 
 		// Store in a new variable for later usage.
 		const newStorageFinal = {...newStorage};
```
Comment (volfkarlo) [inline src/Blocks/components/form/assets/form.js:1093]:
```suggestion
		// Store in a new variable for later usage.
		const newStorageFinal = {...newStorage};
		newStorage.timestamp = Date.now();
```
Wouldn't this avoid the delete process on `newStorageFinal`?

**Review by volfkarlo — CHANGES_REQUESTED** [review]
Mini comments, great job! 👏

File: src/Blocks/components/form/assets/form.js (hunk 3 above)
Comment (iruzevic) [reply src/Blocks/components/form/assets/form.js:1086]: newStorage is used to compare the localStorages and the newStorageFinal is used to have an reference to the original one. This is ok

Comment (volfkarlo) [reply src/Blocks/components/form/assets/form.js:1086]: Yes, but if you change the order, you don't have to delete it.
If you are ok with deleting, all good, this was just a suggestion

**Review by volfkarlo — APPROVED** [review]
👏

**Pull request merged**